frontend/packages/console-shared/src/utils/pod-utils.ts

```diff
@@ -142,3 +146,35 @@ export const isIdled = (deploymentConfig: K8sResourceKind): boolean => {
     false,
   );
 };
+
+export const getPodData = (
+  dc: K8sResourceKind,
+  pods: Pod[],
+  current: PodControllerOverviewItem,
+  previous: PodControllerOverviewItem,
+  isRollingOut: boolean,
+): { inProgressDeploymentData: Pod[] | null; completedDeploymentData: Pod[] | null } => {
```

Comment:
As discussed before, `Pod[]` is a duplicate of `PodKind` with some extra optional properties and Pod is being used in dev console and Podkind is used in the console. fixed the or-type `null` in https://github.com/openshift/console/pull/2496/commits/6353ee79b77029607673e09df07a9099b0fb4a9b

Reply:
I understand that we did talk about it on the refactor PR... but propagating this logic is just going to make it harder to read the code. We can't, as a collective, be referencing historical conversations to explain duplication... otherwise adoption of the code will just grow more complex and difficult to do.

Is there anyway we can covert over to using one or the other (looks like `PodKind` is the official one and `Pod` was a duplicate with smaller # of usages)? Or extending at least so it's traceable. Seeing `current` and `previous` providing a `PodKind[]` that you're just assigning to `Pod[]` like it's the same thing is on the face confusing.

Something like:
```
export interface Pod extends PodKind {
  id?: string;
  name?: string;
}
```
Not sure if `id` and `name` are valid... But at least this way it's more polymorphism and less "typescript deep comparing and not objecting".

Reply:
Agreed! fixed in https://github.com/openshift/console/pull/2496/commits/e4b602178af804b74f431046aa9706db239aadab